File: binding/src/charts/libqvcandlestickmodelmapper.cpp
#include <QCandlestickModelMapper>
#include <QObject>
#include <QString>
#include <QByteArray>
#include <cstring>
#include <type_traits>
#include <QVCandlestickModelMapper>
#include <qvcandlestickmodelmapper.h>
#include "libqvcandlestickmodelmapper.h"
#include "libqvcandlestickmodelmapper.hxx"

QVCandlestickModelMapper* QVCandlestickModelMapper_new() {
	 return new VirtualQVCandlestickModelMapper();
}

QVCandlestickModelMapper* QVCandlestickModelMapper_new2(QObject* parent) {
	 return new VirtualQVCandlestickModelMapper(parent);
}

libqt_string QVCandlestickModelMapper_Tr(const char* s) {
	QString _ret = QVCandlestickModelMapper::tr(s);
	QByteArray _b = _ret.toUtf8();
	libqt_string _str;
	_str.len = _b.length();
	_str.data = static_cast<const char*>(malloc(_str.len + 1));
	memcpy((void*)_str.data, _b.data(), _str.len);
	((char*)_str.data)[_str.len] = '\0';
	return _str;
}

int QVCandlestickModelMapper_Orientation(const QVCandlestickModelMapper* self) {
	return self->orientation();
}

void QVCandlestickModelMapper_SetTimestampRow(QVCandlestickModelMapper* self, int timestampRow) {
	self->setTimestampRow(timestampRow);
}

int QVCandlestickModelMapper_TimestampRow(const QVCandlestickModelMapper* self) {
	return self->timestampRow();
}

void QVCandlestickModelMapper_SetOpenRow(QVCandlestickModelMapper* self, int openRow) {
	self->setOpenRow(openRow);
}

int QVCandlestickModelMapper_OpenRow(const QVCandlestickModelMapper* self) {
	return self->openRow();
}

void QVCandlestickModelMapper_SetHighRow(QVCandlestickModelMapper* self, int highRow) {
	self->setHighRow(highRow);
}

int QVCandlestickModelMapper_HighRow(const QVCandlestickModelMapper* self) {
	return self->highRow();
}

void QVCandlestickModelMapper_SetLowRow(QVCandlestickModelMapper* self, int lowRow) {
	self->setLowRow(lowRow);
}

int QVCandlestickModelMapper_LowRow(const QVCandlestickModelMapper* self) {
	return self->lowRow();
}

void QVCandlestickModelMapper_SetCloseRow(QVCandlestickModelMapper* self, int closeRow) {
	self->setCloseRow(closeRow);
}

int QVCandlestickModelMapper_CloseRow(const QVCandlestickModelMapper* self) {
	return self->closeRow();
}

void QVCandlestickModelMapper_SetFirstSetColumn(QVCandlestickModelMapper* self, int firstSetColumn) {
	self->setFirstSetColumn(firstSetColumn);
}

int QVCandlestickModelMapper_FirstSetColumn(const QVCandlestickModelMapper* self) {
	return self->firstSetColumn();
}

void QVCandlestickModelMapper_SetLastSetColumn(QVCandlestickModelMapper* self, int lastSetColumn) {
	self->setLastSetColumn(lastSetColumn);
}

int QVCandlestickModelMapper_LastSetColumn(const QVCandlestickModelMapper* self) {
	return self->lastSetColumn();
}

void QVCandlestickModelMapper_TimestampRowChanged(QVCandlestickModelMapper* self) {
	self->timestampRowChanged();
}

void QVCandlestickModelMapper_Connect_TimestampRowChanged(QVCandlestickModelMapper* self, intptr_t slot) {
    void (*slotFunc)(QVCandlestickModelMapper*) = reinterpret_cast<void (*)(QVCandlestickModelMapper*)>(slot);
    QVCandlestickModelMapper::connect(self, &QVCandlestickModelMapper::timestampRowChanged, [self, slotFunc]() {
	slotFunc(self);
    });
}

void QVCandlestickModelMapper_OpenRowChanged(QVCandlestickModelMapper* self) {
	self->openRowChanged();
}

void QVCandlestickModelMapper_Connect_OpenRowChanged(QVCandlestickModelMapper* self, intptr_t slot) {
    void (*slotFunc)(QVCandlestickModelMapper*) = reinterpret_cast<void (*)(QVCandlestickModelMapper*)>(slot);
    QVCandlestickModelMapper::connect(self, &QVCandlestickModelMapper::openRowChanged, [self, slotFunc]() {
	slotFunc(self);
    });
}

void QVCandlestickModelMapper_HighRowChanged(QVCandlestickModelMapper* self) {
	self->highRowChanged();
}

void QVCandlestickModelMapper_Connect_HighRowChanged(QVCandlestickModelMapper* self, intptr_t slot) {
    void (*slotFunc)(QVCandlestickModelMapper*) = reinterpret_cast<void (*)(QVCandlestickModelMapper*)>(slot);
    QVCandlestickModelMapper::connect(self, &QVCandlestickModelMapper::highRowChanged, [self, slotFunc]() {
	slotFunc(self);
    });
}

void QVCandlestickModelMapper_LowRowChanged(QVCandlestickModelMapper* self) {
	self->lowRowChanged();
}

void QVCandlestickModelMapper_Connect_LowRowChanged(QVCandlestickModelMapper* self, intptr_t slot) {
    void (*slotFunc)(QVCandlestickModelMapper*) = reinterpret_cast<void (*)(QVCandlestickModelMapper*)>(slot);
    QVCandlestickModelMapper::connect(self, &QVCandlestickModelMapper::lowRowChanged, [self, slotFunc]() {
	slotFunc(self);
    });
}

void QVCandlestickModelMapper_CloseRowChanged(QVCandlestickModelMapper* self) {
	self->closeRowChanged();
}

void QVCandlestickModelMapper_Connect_CloseRowChanged(QVCandlestickModelMapper* self, intptr_t slot) {
    void (*slotFunc)(QVCandlestickModelMapper*) = reinterpret_cast<void (*)(QVCandlestickModelMapper*)>(slot);
    QVCandlestickModelMapper::connect(self, &QVCandlestickModelMapper::closeRowChanged, [self, slotFunc]() {
	slotFunc(self);
    });
}

void QVCandlestickModelMapper_FirstSetColumnChanged(QVCandlestickModelMapper* self) {
	self->firstSetColumnChanged();
}

void QVCandlestickModelMapper_Connect_FirstSetColumnChanged(QVCandlestickModelMapper* self, intptr_t slot) {
    void (*slotFunc)(QVCandlestickModelMapper*) = reinterpret_cast<void (*)(QVCandlestickModelMapper*)>(slot);
    QVCandlestickModelMapper::connect(self, &QVCandlestickModelMapper::firstSetColumnChanged, [self, slotFunc]() {
	slotFunc(self);
    });
}

void QVCandlestickModelMapper_LastSetColumnChanged(QVCandlestickModelMapper* self) {
	self->lastSetColumnChanged();
}

void QVCandlestickModelMapper_Connect_LastSetColumnChanged(QVCandlestickModelMapper* self, intptr_t slot) {
    void (*slotFunc)(QVCandlestickModelMapper*) = reinterpret_cast<void (*)(QVCandlestickModelMapper*)>(slot);
    QVCandlestickModelMapper::connect(self, &QVCandlestickModelMapper::lastSetColumnChanged, [self, slotFunc]() {
	slotFunc(self);
    });
}

libqt_string QVCandlestickModelMapper_Tr2(const char* s, const char* c) {
	QString _ret = QVCandlestickModelMapper::tr(s, c);
	QByteArray _b = _ret.toUtf8();
	libqt_string _str;
	_str.len = _b.length();
	_str.data = static_cast<const char*>(malloc(_str.len + 1));
	memcpy((void*)_str.data, _b.data(), _str.len);
	((char*)_str.data)[_str.len] = '\0';
	return _str;
}

libqt_string QVCandlestickModelMapper_Tr3(const char* s, const char* c, int n) {
	QString _ret = QVCandlestickModelMapper::tr(s, c, n);
	QByteArray _b = _ret.toUtf8();
	libqt_string _str;
	_str.len = _b.length();
	_str.data = static_cast<const char*>(malloc(_str.len + 1));
	memcpy((void*)_str.data, _b.data(), _str.len);
	((char*)_str.data)[_str.len] = '\0';
	return _str;
}

// Base class handler implementation
int QVCandlestickModelMapper_QBaseOrientation(const QVCandlestickModelMapper* self) {
	auto* vqvcandlestickmodelmapper = dynamic_cast<const VirtualQVCandlestickModelMapper*>(self);
	if (vqvcandlestickmodelmapper && vqvcandlestickmodelmapper->isVirtualQVCandlestickModelMapper) {
vqvcandlestickmodelmapper->setQVCandlestickModelMapper_Orientation_IsBase(true);
	return vqvcandlestickmodelmapper->orientation();
}
}

// Auxiliary method to allow providing re-implementation
void QVCandlestickModelMapper_OnOrientation(const QVCandlestickModelMapper* self, intptr_t slot) {
	auto* vqvcandlestickmodelmapper = dynamic_cast<const VirtualQVCandlestickModelMapper*>(self);
	if (vqvcandlestickmodelmapper && vqvcandlestickmodelmapper->isVirtualQVCandlestickModelMapper) {
vqvcandlestickmodelmapper->setQVCandlestickModelMapper_Orientation_Callback(reinterpret_cast<VirtualQVCandlestickModelMapper::QVCandlestickModelMapper_Orientation_Callback>(slot));
}
}

void QVCandlestickModelMapper_Delete(QVCandlestickModelMapper* self) {
    delete self;
}

File: binding/include/charts/libqvcandlestickmodelmapper.zig
const QtC = @import("qt6zig");
const qtc = @import("qt6c");%%_IMPORTLIBS_%%
%%_STRUCTDEFS_%%

/// https://doc.qt.io/qt-6/qvcandlestickmodelmapper.html
pub const qvcandlestickmodelmapper = struct {

    /// New constructs a new QVCandlestickModelMapper object.
    pub fn New() QtC.QVCandlestickModelMapper {
        return qtc.QVCandlestickModelMapper_new();
    }


    /// New2 constructs a new QVCandlestickModelMapper object.
    pub fn New2(parent: ?*anyopaque) QtC.QVCandlestickModelMapper {
        return qtc.QVCandlestickModelMapper_new2(@ptrCast(parent));
    }


    pub fn Tr(s: []const u8, allocator: std.mem.Allocator) []const u8 {
        const _str = qtc.QVCandlestickModelMapper_Tr(@ptrCast(s));
defer qtc.libqt_string_free(&_str);
const _ret = allocator.alloc(u8, _str.len) catch @panic("qvcandlestickmodelmapper.Tr: Memory allocation failed");
@memcpy(_ret, _str.data[0.._str.len]);
return  _ret;
    }

    pub fn Orientation(self: ?*anyopaque, ) i32 {
        return qtc.QVCandlestickModelMapper_Orientation(@ptrCast(self));
    }

    /// Allows for overriding the related default method
    pub fn OnOrientation(self: ?*anyopaque, callback: *const fn (?*anyopaque) callconv(.c) i32) void {
        qtc.QVCandlestickModelMapper_OnOrientation(@ptrCast(self), @intCast(@intFromPtr(callback)));
    }

    /// Base class method implementation
    pub fn QBaseOrientation(self: ?*anyopaque, ) i32 {
        return qtc.QVCandlestickModelMapper_QBaseOrientation(@ptrCast(self));
    }

    pub fn SetTimestampRow(self: ?*anyopaque, timestampRow: i32) void {
        qtc.QVCandlestickModelMapper_SetTimestampRow(@ptrCast(self), timestampRow);
    }

    pub fn TimestampRow(self: ?*anyopaque, ) i32 {
        return qtc.QVCandlestickModelMapper_TimestampRow(@ptrCast(self));
    }

    pub fn SetOpenRow(self: ?*anyopaque, openRow: i32) void {
        qtc.QVCandlestickModelMapper_SetOpenRow(@ptrCast(self), openRow);
    }

    pub fn OpenRow(self: ?*anyopaque, ) i32 {
        return qtc.QVCandlestickModelMapper_OpenRow(@ptrCast(self));
    }

    pub fn SetHighRow(self: ?*anyopaque, highRow: i32) void {
        qtc.QVCandlestickModelMapper_SetHighRow(@ptrCast(self), highRow);
    }

    pub fn HighRow(self: ?*anyopaque, ) i32 {
        return qtc.QVCandlestickModelMapper_HighRow(@ptrCast(self));
    }

    pub fn SetLowRow(self: ?*anyopaque, lowRow: i32) void {
        qtc.QVCandlestickModelMapper_SetLowRow(@ptrCast(self), lowRow);
    }

    pub fn LowRow(self: ?*anyopaque, ) i32 {
        return qtc.QVCandlestickModelMapper_LowRow(@ptrCast(self));
    }

    pub fn SetCloseRow(self: ?*anyopaque, closeRow: i32) void {
        qtc.QVCandlestickModelMapper_SetCloseRow(@ptrCast(self), closeRow);
    }

    pub fn CloseRow(self: ?*anyopaque, ) i32 {
        return qtc.QVCandlestickModelMapper_CloseRow(@ptrCast(self));
    }

    pub fn SetFirstSetColumn(self: ?*anyopaque, firstSetColumn: i32) void {
        qtc.QVCandlestickModelMapper_SetFirstSetColumn(@ptrCast(self), firstSetColumn);
    }

    pub fn FirstSetColumn(self: ?*anyopaque, ) i32 {
        return qtc.QVCandlestickModelMapper_FirstSetColumn(@ptrCast(self));
    }

    pub fn SetLastSetColumn(self: ?*anyopaque, lastSetColumn: i32) void {
        qtc.QVCandlestickModelMapper_SetLastSetColumn(@ptrCast(self), lastSetColumn);
    }

    pub fn LastSetColumn(self: ?*anyopaque, ) i32 {
        return qtc.QVCandlestickModelMapper_LastSetColumn(@ptrCast(self));
    }

    pub fn TimestampRowChanged(self: ?*anyopaque, ) void {
        qtc.QVCandlestickModelMapper_TimestampRowChanged(@ptrCast(self));
    }

    pub fn OnTimestampRowChanged(self: ?*anyopaque, callback: *const fn (?*anyopaque) callconv(.c) void) void {
        qtc.QVCandlestickModelMapper_Connect_TimestampRowChanged(@ptrCast(self), @intCast(@intFromPtr(callback)));
    }

    pub fn OpenRowChanged(self: ?*anyopaque, ) void {
        qtc.QVCandlestickModelMapper_OpenRowChanged(@ptrCast(self));
    }

    pub fn OnOpenRowChanged(self: ?*anyopaque, callback: *const fn (?*anyopaque) callconv(.c) void) void {
        qtc.QVCandlestickModelMapper_Connect_OpenRowChanged(@ptrCast(self), @intCast(@intFromPtr(callback)));
    }

    pub fn HighRowChanged(self: ?*anyopaque, ) void {
        qtc.QVCandlestickModelMapper_HighRowChanged(@ptrCast(self));
    }

    pub fn OnHighRowChanged(self: ?*anyopaque, callback: *const fn (?*anyopaque) callconv(.c) void) void {
        qtc.QVCandlestickModelMapper_Connect_HighRowChanged(@ptrCast(self), @intCast(@intFromPtr(callback)));
    }

    pub fn LowRowChanged(self: ?*anyopaque, ) void {
        qtc.QVCandlestickModelMapper_LowRowChanged(@ptrCast(self));
    }

    pub fn OnLowRowChanged(self: ?*anyopaque, callback: *const fn (?*anyopaque) callconv(.c) void) void {
        qtc.QVCandlestickModelMapper_Connect_LowRowChanged(@ptrCast(self), @intCast(@intFromPtr(callback)));
    }

    pub fn CloseRowChanged(self: ?*anyopaque, ) void {
        qtc.QVCandlestickModelMapper_CloseRowChanged(@ptrCast(self));
    }

    pub fn OnCloseRowChanged(self: ?*anyopaque, callback: *const fn (?*anyopaque) callconv(.c) void) void {
        qtc.QVCandlestickModelMapper_Connect_CloseRowChanged(@ptrCast(self), @intCast(@intFromPtr(callback)));
    }

    pub fn FirstSetColumnChanged(self: ?*anyopaque, ) void {
        qtc.QVCandlestickModelMapper_FirstSetColumnChanged(@ptrCast(self));
    }

    pub fn OnFirstSetColumnChanged(self: ?*anyopaque, callback: *const fn (?*anyopaque) callconv(.c) void) void {
        qtc.QVCandlestickModelMapper_Connect_FirstSetColumnChanged(@ptrCast(self), @intCast(@intFromPtr(callback)));
    }

    pub fn LastSetColumnChanged(self: ?*anyopaque, ) void {
        qtc.QVCandlestickModelMapper_LastSetColumnChanged(@ptrCast(self));
    }

    pub fn OnLastSetColumnChanged(self: ?*anyopaque, callback: *const fn (?*anyopaque) callconv(.c) void) void {
        qtc.QVCandlestickModelMapper_Connect_LastSetColumnChanged(@ptrCast(self), @intCast(@intFromPtr(callback)));
    }

    pub fn Tr2(s: []const u8, c: []const u8, allocator: std.mem.Allocator) []const u8 {
        const _str = qtc.QVCandlestickModelMapper_Tr2(@ptrCast(s), @ptrCast(c));
defer qtc.libqt_string_free(&_str);
const _ret = allocator.alloc(u8, _str.len) catch @panic("qvcandlestickmodelmapper.Tr2: Memory allocation failed");
@memcpy(_ret, _str.data[0.._str.len]);
return  _ret;
    }

    pub fn Tr3(s: []const u8, c: []const u8, n: i32, allocator: std.mem.Allocator) []const u8 {
        const _str = qtc.QVCandlestickModelMapper_Tr3(@ptrCast(s), @ptrCast(c), n);
defer qtc.libqt_string_free(&_str);
const _ret = allocator.alloc(u8, _str.len) catch @panic("qvcandlestickmodelmapper.Tr3: Memory allocation failed");
@memcpy(_ret, _str.data[0.._str.len]);
return  _ret;
    }

    /// Delete this object from C++ memory.
    pub fn QDelete(self: ?*anyopaque) void {
        qtc.QVCandlestickModelMapper_Delete(@ptrCast(self));
    }
};
